aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/DeleteHandler.java

```diff
@@ -27,29 +23,44 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        {
-            throw new CfnInvalidRequestException("Invalid delete request, must contain Primary Id.");
+        if(CollectionUtils.isNullOrEmpty(resourceModel.getTopics())){
+            return ProgressEvent.failed(resourceModel, callbackContext, null, "Value of property Topics must be of type List of String");
         }
 
-        return ProgressEvent.progress(model, callbackContext)
-                .then(progress -> doDelete(proxy, proxyClient, request, progress))
-                .then(progress -> ProgressEvent.<ResourceModel, CallbackContext> builder()
-                        .status(OperationStatus.SUCCESS)
-                        .build());
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Delete TopicPolicy", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress -> Delete(proxy, proxyClient, request, progress, logger))
+                .then(progress -> ProgressEvent.defaultSuccessHandler(null));
     }
 
-    private ProgressEvent<ResourceModel, CallbackContext> doDelete(
+    protected ProgressEvent<ResourceModel, CallbackContext> Delete(
             final AmazonWebServicesClientProxy proxy,
             final ProxyClient<SnsClient> proxyClient,
             final ResourceHandlerRequest<ResourceModel> request,
-            final ProgressEvent<ResourceModel, CallbackContext> progress) {
-        List<String> topics = request.getDesiredResourceState().getTopics();
-        return handleDelete(proxy, proxyClient, request, progress, topics, ACTION_DELETED, DELETE_HANDLER , logger);
+            ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+        final CallbackContext callbackContext = progress.getCallbackContext();
+        List<String> topics = model.getTopics();
+        for (final String topicArn : topics) {
+            String policy = getDefaultPolicy(request,topicArn);
+            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy
+                    .initiate("AWS-SNS-TopicPolicy::Delete", proxyClient, model, callbackContext)
+                    .translateToServiceRequest((resourceModel) -> Translator.translateToRequest(topicArn, policy))
+                    .makeServiceCall((awsRequest, client) -> {
+                        SetTopicAttributesResponse response = proxyClient.injectCredentialsAndInvokeV2(awsRequest, client.client()::setTopicAttributes);
+                        logger.log(String.format("Resource Deleted in StackId: %s", request.getStackId()));
+                        return response;
+                    })
+                    .handleError((awsRequest, exception, client, rModel, context) -> handleError(awsRequest, exception, client, rModel, context))
+                    .success();
```

Comment:
before returning success we need to handle this propagation delay from this function: https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/topicPolicy/TopicPolicyDeleteWorkflowImpl.java#:~:text=PropertyResourceDeleteHandler%20deleteHandler%20%3D%20new%20PropertyResourceDeleteHandler(propertyResourceLockActivitiesClient%2C%2030%2C%20Duration.standardSeconds(30))

it adds a 30 second propagation delay and calls deleteHandler.delete so we need to implement that before returning success: https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/topicPolicy/TopicPolicyDeleteWorkflowImpl.java#:~:text=waitFor%20%3D%20deleteHandler.delete(getResourceType()%2C%20topicArn%2C%20getStackId()%2C%20getLogicalResourceId()%2C%20waitFor)%3B

Reply:
This delay related to Property Resource Lock. Skip for now
https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resourceLock/PropertyResourceCreateHandler.java

Reply:
... Why are you guys putting internal GitFarm link in GH? Could you remove your comments?